### PR TITLE
Move filter.notify to WakuRelay handler and add test

### PR DESCRIPTION
**Pull request by kdeme**
Fixing https://github.com/status-im/nim-waku/issues/114 and adding test cases for WakuNode.

Could use further clean-up on the Filters part and a default relayHandler should probably be available for WakuNode
Also need to implement still the key part of the Filters table.

### CODE REVIEW

File: waku/node/v2/rpc/wakurpc.nim [hunk 1]
```diff
@@ -40,6 +40,7 @@ proc setupWakuRPC*(node: WakuNode, rpcsrv: RpcServer) =
     proc handler(topic: string, data: seq[byte]) {.async, gcsafe.} =
       info "Hit subscribe handler", topic=topic, data=data
 
+    # TODO: Shouldn't we really be doing WakuNode subscribe here?
```
Comment (oskarth) [inline waku/node/v2/rpc/wakurpc.nim:43]: Yeah we should, this is from before node API existed and wasn't updated correctly.

File: waku/node/v2/rpc/wakurpc.nim [hunk 2]
```diff
@@ -26,8 +26,8 @@ proc setupWakuRPC*(node: WakuNode, rpcsrv: RpcServer) =
     # XXX Why is casting necessary here but not in Nim node API?
     let wakuRelay = cast[WakuRelay](node.switch.pubSub.get())
     # XXX also future return type
-    var message = WakuMessage(payload: payload, contentTopic: "foo")
-    discard wakuRelay.publish(topic, message)
+    # TODO: Shouldn't we really be doing WakuNode publish here?
```
Comment (oskarth) [inline waku/node/v2/rpc/wakurpc.nim:29]: Ditto below

**Review by oskarth — APPROVED** [review]

**Comment by decanus** [comment]
@oskarth you used admin override to merge, please don't do that next time. It messed up a whole bunch of stuff.

**Comment by oskarth** [comment]
What did it mess up? CI sometimes broken, and this had passed Ubuntu build/test phase.

**Pull request merged**